### ISSUE TITLE
Multiple local externals possible?

### ISSUE THREAD

**Issue by mathomp4** [Fortran-FOSS-Programmers/ford#468]
I'm trying to figure out if you can have multiple local externals. 

I tried doing:
```
external: local = Ford/doc-esmf
          local = Ford/doc-pFUnit
          local = Ford/doc-pFlogger
          local = Ford/doc-fArgParse
```

but my job is throwing:
```
RuntimeError: Could not register macro "|local|" as "Ford/doc-pFUnit" because it is already defined as "Ford/doc-esmf".
```

**Comment by ZedThree** [comment]
This isn't documented very well, sorry. Yes, it is possible, the `local`/`remote` names in the external projects test are just names and don't affect how ford deals with them.

Try:

```
external: esmf = Ford/doc-esmf
          pFUnit = Ford/doc-pFUnit
          pFlogger = Ford/doc-pFlogger
          fArgParse = Ford/doc-fArgParse
```

which should register `|esmf|`, `|pFUnit|`, etc. as macros that expand to the respective project URLs (or file locations in this case)

**Comment by mathomp4** [comment]
@ZedThree Ah thanks. I'll give it a try.

However, it turns out that in the end I don't think I can do what I wanted to do. For some reason that seems obviously dumb to me now, I'd hoped I could generate all those external docs during a GitHub Action and then use it as "extra" docs for my main MAPL documentation (rather than depending on others to make Ford docs and host them themselves). 

Of course, the issue is that local docs are pushed up by [github-pages-deploy-action](https://github.com/JamesIves/github-pages-deploy-action) by @JamesIves and I think it can only see a single folder. 

I wonder if I can some how "embed" the ESMF docs inside my MAPL docs? Hmm...possible?

**Comment by mathomp4** [comment]
Confirmed if I don't use `local` 4 times it works. 😄 

I also have sort of figured out how I could possible do what I was thinking if I did:

```
doc/MAPL
doc/extern/esmf
doc/extern/yaFyaml
...
```

I had to put MAPL in a subdir because I build the externals first then MAPL and otherwise Ford was nuking `doc/` and no externs! And Ford is pretty clear [about how it works](https://forddocs.readthedocs.io/en/latest/user_guide/project_file_options.html#output-dir):

> output_dir
> 
> The directory where the project output will be placed. **Any content already present there will be deleted.** (default: ./doc)

And it works just fine when reading from local disk. But, I'm not sure how to get the deploy action to like this structure as it's a miracle I figured out what I've currently done (lots of trial and error and copying what others have done). I'm sure there some way with [maybe `target-folder` or something](https://github.com/JamesIves/github-pages-deploy-action#optional-choices), but...eh. I'll probably just work on getting actual docs sites online in all the GFE libs and then it's simple.
